server/channels/web/handlers_test.go

```diff
@@ -927,3 +930,156 @@ func TestGetOriginClient(t *testing.T) {
 		require.Equal(t, tc.expectedClient, actualClient)
 	}
 }
+
+func noOpHandler(_ *Context, _ http.ResponseWriter, _ *http.Request) {
+	// no op
+}
+
+func TestHandlerServeHTTPBasicSecurityChecks(t *testing.T) {
+	t.Run("Should not cause 414 error if url is smaller than configured limit", func(t *testing.T) {
+		th := SetupWithStoreMock(t)
+		defer th.TearDown()
+
+		web := New(th.Server)
+		handler := web.NewHandler(noOpHandler)
+
+		// using the default URL length
+		request := httptest.NewRequest("GET", "/api/v4/test?with=not&many=query_params", nil)
+		response := httptest.NewRecorder()
+		handler.ServeHTTP(response, request)
+		assert.Equal(t, http.StatusOK, response.Code)
+	})
+
+	t.Run("Should cause 414 error if url is smaller than configured limit", func(t *testing.T) {
```

Comment:
if url is longer than ..

Reply:
Imaging how much energy CI servers consume globally re-running the job after a typo fix 🤦

Reply:
There has been efforts to persist the test cache across CI runs. So it should only re-run tests for the package which changed. I had taken a stab at that but somehow couldn't get it work. Need to spend more time. I'd love it if you want to take a look.

Context: https://community.mattermost.com/core/pl/ewftajr5rib4bbjqqw9i1putoc